src/app/pages/add-course-page/add-course-page.component.ts

```diff
@@ -20,21 +23,20 @@ export class AddCoursePageComponent implements OnInit {
     private coursesService: CoursesService,
     private activatedRouter: ActivatedRoute,
     private router: Router,
+    private store: Store,
     ) {
   }
 
   ngOnInit(): void {
     const id = parseInt(this.activatedRouter.snapshot.params.id, 10);
-    this.courses = this.coursesService.courses;
+    this.courses = this.store.pipe(select(selectCourses));
     this.coursesService.loading.subscribe(state => this.loading = state);
-
     if (id) {
       this.isEdit = true;
-      this.coursesService.courses.subscribe(courses => {
-          this.course = courses.find(item => item.id === id);
+      this.courses.subscribe(courses => {
```

Comment:
You don't need to subscribe on courses here.
You need to add LoadSelectedCourse and CreateNewCourse actions and dispatch this courses based on the existing ID.